blockchain/utxoviewpoint.go

```diff
@@ -54,7 +54,7 @@ func (view *UtxoViewpoint) LookupEntry(outpoint wire.OutPoint) *UtxoEntry {
 // marked unspent.  All fields will be updated for existing entries since it's
 // possible it has changed during a reorg.
 func (view *UtxoViewpoint) addTxOut(outpoint wire.OutPoint, txOut *wire.TxOut,
-	packedFlags utxoFlags, blockHeight int64, blockIndex uint32,
+	state utxoState, packedFlags utxoFlags, blockHeight int64, blockIndex uint32,
```

Comment:
Perhaps there is a reason for this in a later commit that I'm not aware of, but it seems to me that the very act of adding a new output to the view dictates its state will be modified and unspent.

Given that, rather than passing in the state, what do you think about just modifying the entry state accordingly once in here (down where it currently does `entry.state = state`) instead of doing it at the call sites?

Something like the following to match `disconnectTransactions`:

```Go
// Mark the existing referenced transaction output as unspent and modified.
entry.state &^= utxoStateSpent
entry.state |= utxoStateModified
```

Reply:
This function is only getting called from `AddTxOut` and `AddTxOuts` which always pass in the state as `utxoStateModified`.  I think it's a good call just to set it in this function.  Updated as suggested.